src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -960,6 +963,45 @@ void Terminal::_NotifyTerminalCursorPositionChanged() noexcept
     }
 }
 
+// Method Description:
+// - Implements the logic to determine if a location is in a given region
+// Arguments:
+// - The location
+// - The start and end coordinates of the region
+// Return value:
+// - True if the location is within those coordinates, false otherwise
+bool Microsoft::Terminal::Core::Terminal::_IsLocationWithinCoordinates(const COORD location, const COORD first, const COORD second) const noexcept
+{
+    if (first.Y == second.Y)
+    {
+        const auto sameRow = location.Y == first.Y;
+        const auto inRange = (first.X <= location.X && location.X < second.X);
+        return (sameRow && inRange);
+    }
+    else
+    {
+        // check first row
+        if (location.Y == first.Y && (first.X <= location.X))
+        {
+            return true;
+        }
+        // check rows in between first row and last row
+        for (auto curRow = first.Y + 1; curRow < second.Y; ++curRow)
+        {
+            if (location.Y == curRow)
+            {
+                return true;
+            }
+        }
+        // check last row
+        if (location.Y == second.Y && location.X < second.X)
+        {
+            return true;
+        }
+    }
+    return false;
+}

```

Comment:
I SWEAR this function already exists in utils, types, or TIL already. I can help you try to find it if you need.

Reply:
Switched to interval tree implementation so this is gone now